File: Labs/Lab 5/lab5.py
class TreeNode(object):

    def __init__(self, val=0, left=None, right=None):
        self.val = val

        self.left = left

        self.right = right

def isValidSubtree(root, min, max):
    if root == None:
        return True
    if root.val <= min or root.val >= max:
        return False
    return isValidSubtree(root.left, min, root.val) and isValidSubtree(root.right, root.val, max)


def isValidBST(self, root):
    return isValidSubtree(root, float('-inf'), float('inf'))

File: Labs/Lab 5/Lab5.java
package lab5;

public class Lab5 {

	public boolean isValidBST(TreeNode root) {
		return isValidSubtree(root, Integer.MIN_VALUE, Integer.MAX_VALUE);
		
	 }
	
	
	public boolean isValidSubtree(TreeNode root, int min, int max) {
		if(root==null) return true;
		
		if(root.val<=min || root.val>=max) return false;
		
		return isValidSubtree(root.left, min, root.val) && 
				isValidSubtree(root.right, root.val, max);

	}
	
}
